src/test/fuzz/util.h

```diff
@@ -534,4 +552,118 @@ void ReadFromStream(FuzzedDataProvider& fuzzed_data_provider, Stream& stream) no
     }
 }
 
+class FuzzedSock : public Sock
+{
+    FuzzedDataProvider& m_fuzzed_data_provider;
+
+public:
+    explicit FuzzedSock(FuzzedDataProvider& fuzzed_data_provider) : m_fuzzed_data_provider{fuzzed_data_provider}
+    {
+    }
+
+    ~FuzzedSock() override
+    {
+    }
+
+    SOCKET Get() const override
+    {
+        assert(false && "Not implemented yet.");
+    }
+
+    SOCKET Release() override
+    {
+        assert(false && "Not implemented yet.");
+    }
+
+    void Reset() override
+    {
+        assert(false && "Not implemented yet.");
+    }
+
+    ssize_t Send(const void* data, size_t len, int flags) const override
+    {
+        const std::vector<int> send_errnos{
+            EACCES,
+            EAGAIN,
+            EALREADY,
+            EBADF,
+            ECONNRESET,
+            EDESTADDRREQ,
+            EFAULT,
+            EINTR,
+            EINVAL,
+            EISCONN,
+            EMSGSIZE,
+            ENOBUFS,
+            ENOMEM,
+            ENOTCONN,
+            ENOTSOCK,
+            EOPNOTSUPP,
+            EPIPE,
+            EWOULDBLOCK,
+        };
+        if (m_fuzzed_data_provider.ConsumeBool()) {
+            return len;
+        }
+        const ssize_t r = m_fuzzed_data_provider.ConsumeIntegralInRange<ssize_t>(-1, len);
+        if (r == -1) {
+            SetFuzzedErrNo(m_fuzzed_data_provider, send_errnos);
+        }
+        return r;
+    }
+
+    ssize_t Recv(void* buf, size_t len, int flags) const override
+    {
+        const std::vector<int> recv_errnos{
+            EAGAIN,
+            EBADF,
+            ECONNREFUSED,
+            EFAULT,
+            EINTR,
+            EINVAL,
+            ENOMEM,
+            ENOTCONN,
+            ENOTSOCK,
+            EWOULDBLOCK,
+        };
+        assert(buf != nullptr || len == 0);
+        if (len == 0 || m_fuzzed_data_provider.ConsumeBool()) {
+            const ssize_t r = m_fuzzed_data_provider.ConsumeBool() ? 0 : -1;
+            if (r == -1) {
+                SetFuzzedErrNo(m_fuzzed_data_provider, recv_errnos);
+            }
+            return r;
+        }
+        const std::vector<uint8_t> random_bytes = m_fuzzed_data_provider.ConsumeBytes<uint8_t>(m_fuzzed_data_provider.ConsumeBool() ? len : m_fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, len));
```

Comment:
nit: Would be nice to break long lines after each `?`, or `:`

Reply:
Fixed!